### PR TITLE
Run regresscheck-t on PG13

### PR DESCRIPTION
**Pull request by svenklemm**
This patch enables running regresscheck-t on PG13

### CODE REVIEW

**Review by mkindahl — APPROVED** [review]
Changes look fine. There is a weird error in the 13.1 regression test though.

File: tsl/src/continuous_aggs/create.c [hunk 1]
```diff
@@ -1514,11 +1513,11 @@ finalizequery_get_select_query(FinalizeQueryInfo *inp, List *matcollist,
 		Value *attrname = makeString(cdef->colname);
 		rte->eref->colnames = lappend(rte->eref->colnames, attrname);
 	}
+	rte->requiredPerms |= ACL_SELECT;
+	rte->selectedCols = bms_add_member(rte->selectedCols, 0 - FirstLowInvalidHeapAttributeNumber);
```
Comment (gayyappan) [inline tsl/src/continuous_aggs/create.c:1517]: should move this to inside the loop at line 1511 and add all the columns.

File: tsl/test/expected/transparent_decompression_ordered_index-13.out [hunk 2]
```diff
@@ -0,0 +1,979 @@
...
+---------------------------------------------------------------------------------------------------------------------------------------------------------------
+ Sort (actual rows=48 loops=1)
+   Sort Key: mt_1."time"
+   Sort Method: quicksort 
+   ->  Nested Loop (actual rows=48 loops=1)
+         Join Filter: ((mt_1."time" > nd.start_time) AND (mt_1."time" < nd.stop_time) AND (mt_1.device_id = nd.node))
+         Rows Removed by Join Filter: 1493
+         ->  Append (actual rows=1541 loops=1)
+               ->  Custom Scan (DecompressChunk) on _hyper_1_1_chunk mt_1 (actual rows=480 loops=1)
+                     ->  Index Scan using compress_hyper_2_6_chunk__compressed_hypertable_2_device_id__ts on compress_hyper_2_6_chunk (actual rows=5 loops=1)
```
Comment (gayyappan) [inline tsl/test/expected/transparent_decompression_ordered_index-13.out:458]: Looks like the filters don't get applied at decompresschunk anymore and moved to the join. Also no indexcond for the index scan

File: tsl/test/expected/transparent_decompression-13.out [hunk 3]
```diff
@@ -0,0 +1,9902 @@
...
+   ->  Custom Scan (ChunkAppend) on public.metrics (actual rows=0 loops=1)
+         Output: metrics."time", metrics.device_id, metrics.device_id_peer, metrics.v0, metrics.v1, metrics.v2, metrics.v3
+         Order: metrics."time"
+         Startup Exclusion: false
+         Runtime Exclusion: false
+         ->  Sort (actual rows=0 loops=1)
+               Output: _hyper_1_1_chunk."time", _hyper_1_1_chunk.device_id, _hyper_1_1_chunk.device_id_peer, _hyper_1_1_chunk.v0, _hyper_1_1_chunk.v1, _hyper_1_1_chunk.v2, _hyper_1_1_chunk.v3
+               Sort Key: _hyper_1_1_chunk."time"
+               Sort Method: quicksort 
+               ->  Custom Scan (DecompressChunk) on _timescaledb_internal._hyper_1_1_chunk (actual rows=0 loops=1)
```
[the remark this answers is not in the document]
Comment (gayyappan) [reply tsl/test/expected/transparent_decompression-13.out:467]: wrong file. closing comment.

File: tsl/test/expected/jit-13.out [hunk 4]
```diff
@@ -0,0 +1,225 @@
+-- This file and its contents are licensed under the Timescale License.
```
Comment (gayyappan) [inline tsl/test/expected/jit-13.out:1]: Not able to get github to display this file correctly. Summarizing the changes here:
 at line 2317:  we have plans without indexcond. I think this is similar to the other diffs seen int transparent_dcompression_ordered_index.out

File: tsl/src/continuous_aggs/create.c [hunk 5]
```diff
@@ -1513,12 +1512,14 @@ finalizequery_get_select_query(FinalizeQueryInfo *inp, List *matcollist,
 		ColumnDef *cdef = (ColumnDef *) lfirst(lc);
 		Value *attrname = makeString(cdef->colname);
 		rte->eref->colnames = lappend(rte->eref->colnames, attrname);
+		rte->selectedCols =
+			bms_add_member(rte->selectedCols,
```
Comment (gayyappan) [inline tsl/src/continuous_aggs/create.c:1516]: this should be bms_add_member(rte->selectedCols, attno)?

Comment (gayyappan) [reply tsl/src/continuous_aggs/create.c:1516]: is it possible to use bms_add_members to add them all together?

Comment (svenklemm) [reply tsl/src/continuous_aggs/create.c:1516]: the numbers in the bitmapset are offset to allow for systemcolumns to be represented as well. bitmapsets can only store positive numbers but systemcolumns are negative so postgres uses an offset here to be able to represent them as well.

**Review by gayyappan — APPROVED** [review]
Plan changes that no longer push down filters inside decompresschunk will be investigated separately. The other changes look good.

**Pull request merged**